Assets/Scripts/Effects/FogOfSanity.cs

```diff
@@ -0,0 +1,100 @@
+﻿using Entity.Player;
+using UnityEngine;
+
+
+public class FogOfSanity : MonoBehaviour
+{
+    [SerializeField][Tooltip("Reference to the fog generating mesh")]
+    private Renderer fogOfSanityMesh;
+    
+    [SerializeField][Tooltip("Reference to the fog generating mesh")]
+    private Camera mainCamera;
+
+    [SerializeField] [Tooltip("Intensity of the pulse animation")]
+    private float pulseIntensity;
+    
+    [SerializeField] [Tooltip("Determines how fast pulse animation is")]
+    private float pulseFrequency;
+
+    // radius of the fog
+    private float _baseRadius;
+    
+    // properties needed for the pulse animation
+    private float _currentRadius;
+    private bool _isGrowing;
+   
+    // shader properties
+    private static readonly int PlayerPosition = Shader.PropertyToID("_PlayerPosition");
+    private static readonly int FogRadius = Shader.PropertyToID("_FogRadius");
+
+    private void Awake()
+    {
+        _baseRadius = 500;
```

Comment:
Do we need to change this value depending on the screensize?

Reply:
this value is relative to clip space coordinates, so i'd say no. But i'd have to jump deeper into shader logic to say for certain. However it's scaling mechanic is pretty basic at the moment and will need some fine tuning, once we completed all the visual stuff that is connected to the sanity system anyway.